src/backend/distributed/commands/collation.c

```diff
@@ -63,24 +63,62 @@ CreateCollationDDLInternal(Oid collationId, Oid *collowner, char **quotedCollati
 	const char *collname = NameStr(collationForm->collname);
 	bool collisdeterministic = collationForm->collisdeterministic;
 
+	char *collcollate;
+	char *collctype;
+
 #if PG_VERSION_NUM >= PG_VERSION_15
+
+	/*
+	 * In PG15, there is an added option to use ICU as global locale provider.
+	 * pg_collation has three locale-related fields: collcollate and collctype,
+	 * which are libc-related fields, and a new one colliculocale, which is the
+	 * ICU-related field. Only the libc-related fields or the ICU-related field
+	 * is set, never both.
+	 */
+	char *colliculocale;
 	bool isnull;
+
 	Datum datum = SysCacheGetAttr(COLLOID, heapTuple, Anum_pg_collation_collcollate,
 								  &isnull);
-	Assert(!isnull);
-	char *collcollate = TextDatumGetCString(datum);
+	if (!isnull)
+	{
+		collcollate = TextDatumGetCString(datum);
+	}
+	else
+	{
+		collcollate = NULL;
+	}
+
 	datum = SysCacheGetAttr(COLLOID, heapTuple, Anum_pg_collation_collctype, &isnull);
-	Assert(!isnull);
-	char *collctype = TextDatumGetCString(datum);
+	if (!isnull)
+	{
+		collctype = TextDatumGetCString(datum);
+	}
+	else
+	{
+		collctype = NULL;
+	}
+
+	datum = SysCacheGetAttr(COLLOID, heapTuple, Anum_pg_collation_colliculocale, &isnull);
+	if (!isnull)
+	{
+		colliculocale = TextDatumGetCString(datum);
```

Comment:
Don't we need `pstrdup` for PG 15 part?

Reply:
No, because in PG15, collcollate and collctype are of type text, not name.